docker.py

```diff
@@ -42,75 +75,141 @@ def main(tarball, install_iu, uninstall_iu):
     installed = prefix/'ccs'
 
     for parent in reversed(pathlib.Path(setup).parents):
-        print(list(parent.iterdir()))
+        print_flush(list(parent.iterdir()))
 
-    try:
-        print('setup:', setup)
-        subprocess.run(
-            [
-                setup,
-                '--unattendedmodeui', 'none',
-                '--mode', 'unattended',
-                '--enable-components', 'PF_C28',
-                '--prefix', os.fspath(prefix),
-            ],
-            check=True
-        )
-        print('succcesss--------------------------------')
-    except subprocess.CalledProcessError:
-        install_logs = installed/'install_logs'
-        for parent in install_logs.parents:
-            print('checking: ', parent, parent.exists())
-        install_logs = list(
-            path
-            for path in install_logs.rglob('*')
-            if path.is_file()
-        )
-        print('potential install logs:')
-        for maybe in install_logs:
-            print('   ', maybe)
-        for install_log in install_logs:
-            print('    --------: {}'.format(install_log))
-            with open(install_log) as f:
-                for line in f:
-                    sys.stdout.write(line)
-
-        raise
-
-    ccstudio = installed/'eclipse'/'ccstudio'
-
-    for parent in reversed(ccstudio.parents):
-        print(list(parent.iterdir()))
-
-    link = pathlib.Path(os.sep)/'usr'/'local'/'bin'/'ccstudio'
-    link.symlink_to(ccstudio)
-
-    iu_to_install = install_iu
-    if len(iu_to_install) > 0:
-        print('Installing IU {}'.format(iu_to_install))
-        install_iu(iu=iu_to_install, ccstudio=ccstudio)
-    else:
-        print('No IU specified for installation')
-
-    iu_to_uninstall = uninstall_iu
-    if len(iu_to_uninstall) > 0:
-        print('Uninstalling IU {}'.format(iu_to_uninstall))
-        uninstall_iu(iu=iu_to_uninstall, ccstudio=ccstudio)
-    else:
-        print('No IU specified for uninstallation')
+    with virtual_display():
+        print_flush('+++ here')
+        try:
+            print_flush('setup:', setup)
+            subprocess.run(
+                [
+                    setup,
+                    '--unattendedmodeui', 'none',
+                    '--mode', 'unattended',
+                    '--enable-components', 'PF_C28',
+                    '--prefix', os.fspath(prefix),
+                ],
+                check=True
+            )
+            print_flush('success--------------------------------')
+        except subprocess.CalledProcessError:
+            install_logs = installed/'install_logs'
+            for parent in install_logs.parents:
+                print_flush('checking: ', parent, parent.exists())
+            install_logs = list(
+                path
```

Comment:
formatting, should this all be on one line?

Reply:
i'll read that as "start using black!".  `:]`  (and change it for now)  (but no, black wouldn't put it on one line, though it also wouldn't format it per my preference which is what is there)